### PR TITLE
Propagate native references to merged project wrapper outputs

### PR DESCRIPTION
**Pull request by trylek**
When building tests in merged mode, we need to copy the native
references along with the component test dlls to the output folder
to make them discoverable by the OS loader.

Thanks

Tomas

### CODE REVIEW

File: src/tests/Directory.Build.targets
Comment (jkoritzinsky) [inline src/tests/Directory.Build.targets]: It doesn't look like this property is used. Do we need to set it?
```suggestion
```

**Review by jkoritzinsky — APPROVED** [review]
I think I'd prefer that we'd rather hook into the `GetCopyToOutputDirectoryItems` infrastructure instead of this design, but I'm okay with this design for now as it's simple and will get us off the ground.

File: src/tests/Directory.Build.targets
Comment (trylek) [reply src/tests/Directory.Build.targets]: Thanks Jeremy for spotting, I have removed the superfluous property I originally used as local instrumentation when debugging the change in 2nd commit on this PR.

Comment (trylek) [reply src/tests/Directory.Build.targets]: If you prefer triggering this action off another build target or rolling in into an existing one, I have no problems with it whatsoever, I have actually only started testing this change on a larger scale and I have mostly randomly put the code next to the existing logic for publishing native product items.

Comment (trylek) [reply src/tests/Directory.Build.targets]: (Just for context, this accounts for the two remaining failures apart from the six ones caused by the ilproj rename logic.)

Comment (jkoritzinsky) [reply src/tests/Directory.Build.targets]: I think leaving it as-is for this change just to get us off the ground is reasonable.

We have a lot of complexity around how we support copying the native binaries around (due to the fact that we can do the copy as a separate step from the build), so I'm fine holding off doing a more invasive change and just getting in something simple to unblock.

**Pull request merged**
